### PR TITLE
feat: update edit source datasets ui general information section for published (#79)

### LINKED ISSUE #79: Update edit source datasets UI general information section for published
As a starting point, we can add:

image

#### DOD
1. But without the tab for "Unpublished"
2. DOI is not editable
3. Title is not editable
4. Green dot indicates crossref status "OK"
5. no vertical dot thingo
6. no refresh button
7. So no save or discard buttons as nothing is editable!

### PR DESCRIPTION
**Pull request by frano-m**
Closes #79.

Includes minor refactor of form field constants i.e. field names and default input props. Facilitates simplification of rendering "shared" controllers between e.g. "Add" / "Edit" forms where prop values such as "placeholder" or "label" can be updated from a single source.

Also includes the removal of components no longer in use.

### CODE REVIEW

File: app/apis/catalog/hca-atlas-tracker/common/utils.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export function dbSourceDatasetToApiSourceDataset(
 }
 
 export function getAtlasName(atlas: HCAAtlasTrackerAtlas): string {
-  return `${atlas.shortName} ${atlas.version}`;
+  return `${atlas.shortName} v${atlas.version}`;
```
Comment (frano-m) [inline app/apis/catalog/hca-atlas-tracker/common/utils.ts:76]: Prefix `v` to version...

File: app/components/Detail/components/AddSourceDataset/addSourceDataset.tsx [hunk 2]
```diff
@@ -15,8 +15,8 @@ import {
   BUTTON_COLOR,
 } from "../../../common/Button/components/ButtonLink/buttonLink";
 import { Divider } from "../TrackerForm/components/Divider/divider.styles";
-import { AuthenticationRequired } from "../TrackerForm/components/Section/components/Atlas/components/AuthenticationRequired/authenticationRequired";
-import { GeneralInfo } from "../TrackerForm/components/Section/components/SourceDataset/components/GeneralInfo/generalInfo";
+import { AuthenticationRequired } from "../TrackerForm/components/Section/components/AuthenticationRequired/authenticationRequired";
+import { GeneralInfo } from "../TrackerForm/components/Section/components/SourceDataset/components/Add/components/GeneralInfo/generalInfo";
```
Comment (frano-m) [inline app/components/Detail/components/AddSourceDataset/addSourceDataset.tsx:19]: GeneralInfo component are now created for "Add" and "Edit" views.

File: app/components/Detail/components/TrackerForm/components/Section/components/Atlas/components/GeneralInfo/generalInfo.tsx [hunk 3]
```diff
@@ -48,85 +44,71 @@ export const GeneralInfo = ({
...
-                label="Short name"
-                placeholder="e.g. Cortex"
-                readOnly={false}
-              />
-            );
-          }}
+          name={FIELD_NAME.SHORT_NAME}
+          render={({ field }): JSX.Element => (
+            <Input
+              {...field}
```
Comment (frano-m) [inline app/components/Detail/components/TrackerForm/components/Section/components/Atlas/components/GeneralInfo/generalInfo.tsx:50]: Updated field name constant to `SHORT_NAME`, and used new `DEFAULT_INPUT_PROPS` here and elsewhere to define certain props shared across "Add" / "Edit" views.

File: app/views/AddNewSourceDatasetView/addNewSourceDatasetView.tsx [hunk 4]
```diff
@@ -18,11 +17,7 @@ export const AddNewSourceDatasetView = ({
   const formMethod = useAddSourceDatasetForm();
   return (
     <DetailView
-      breadcrumbs={
-        <Breadcrumbs
-          breadcrumbs={getBreadcrumbs(atlasId, atlas && getAtlasName(atlas))}
-        />
-      }
+      breadcrumbs={<Breadcrumbs breadcrumbs={getBreadcrumbs(atlasId, atlas)} />}
```
Comment (frano-m) [inline app/views/AddNewSourceDatasetView/addNewSourceDatasetView.tsx:20]: Moved `getAtlasName` into `getBreadcrumbs` to make the arguments more generic...

File: app/views/EditSourceDatasetView/hooks/useEditSourceDatasetForm.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * Returns schema default values mapped from source dataset.
+ * @param sourceDataset - Source dataset.
+ * @returns schema default values.
+ */
+function mapSchemaValues(
+  sourceDataset?: HCAAtlasTrackerSourceDataset
+): SourceDatasetEditData {
+  return {
+    [FIELD_NAME.CITATION]: getSourceDatasetCitation(sourceDataset),
```
[the remark this answers is not in the document]
Comment (frano-m) [reply app/views/EditSourceDatasetView/hooks/useEditSourceDatasetForm.tsx:33]: See [general info section -> publication field -> help text](https://www.figma.com/file/i4uZsYkKAML9ivi2TIo6ug/HCA-Data-Explorer?type=design&node-id=6827-10110&mode=design&t=AjSGjZVaZs7qLL2D-4)

File: site-config/hca-atlas-tracker/local/index/atlasEntityConfig.ts
Comment (frano-m) [inline site-config/hca-atlas-tracker/local/index/atlasEntityConfig.ts:1]: Dave requested we remove Version from the Atlas table, as the value is already under the Name column

**Review by hunterckx — APPROVED** [review]
Looks good to me!

**Pull request merged**